rs/boundary_node/ic_boundary/src/core.rs

```diff
@@ -210,6 +230,76 @@ pub async fn main(cli: Cli) -> Result<(), Error> {
         cli.rate_limiting.rate_limit_generic.clone(),
     ));
 
+    // HTTP Logs Anonymization
+    let salt = Arc::new(ArcSwapOption::<Vec<u8>>::empty());
+
+    // Vault
+    let v = vault_from_config(
+        &CryptoConfig::new("/home/orricon/workspace/ic-cdk-logs/out/node-1/crypto".into()), // config
+        None,                            // tokio_runtime_handle
+        no_op_logger(),                  // logger
+        Arc::new(CryptoMetrics::none()), // metrics
+    );
+
+    let pk = v
+        .current_node_public_keys()
+        .map_err(|err| anyhow!("failed to retrieve public key: {err:?}"))?
+        .node_signing_public_key
+        .context("missing node public key")?;
+
+    let pk = CspPublicKey::Ed25519(pk.key_value.try_into()?);
+
+    let algo = pk.algorithm_id();
+    let kid = KeyId::try_from(&pk).map_err(|err| anyhow!("failed to derive key id: {err:?}"))?;
+
+    // Custom Signer
+    let s = Sender::Node {
+        pub_key: pk.pk_bytes().to_vec(),
+        sign: Arc::new(move |msg: &MessageId| {
+            let sig = v
+                .sign(
+                    algo,                  // algorithm_id
+                    msg.as_signed_bytes(), // message
+                    kid,                   // key_id
+                )
+                .map_err(|err| anyhow!("failed to sign message: {err:?}"))?;
+
+            Ok(sig.as_ref().to_vec())
+        }),
+    };
+
+    // Agent
+    let (proto, port) = [
+        // HTTP
+        cli.listen.listen_http_port.map(|p| ("http", p)),
+        // HTTPS
+        #[cfg(feature = "tls")]
+        cli.listen.listen_https_port.map(|p| ("https", p)),
+    ]
+    .into_iter()
+    // Choose first non-none option
+    .flatten()
+    .next()
+    .ok_or(anyhow!("missing http and https listen ports"))?;
+
+    let ag = Agent::new(
+        format!("{proto}://localhost:{port}").parse()?, // url
+        s,                                              // sender
+    );
+
+    // Canister
+    let c = AnonymizationCanister::new(
+        ag,                     // agent
+        *LOG_ANONYMIZATION_CID, // canister_id
+    );
+
+    // Rng
+    let rng_seed = SystemTime::now().duration_since(UNIX_EPOCH)?.as_nanos();
```

Comment:
This isn't secure... if that's intentional there should probably be a comment along those lines.

Reply:
Hey @randombit, perhaps you can help me figure out _if_ we need a secure RNG here? Basically this RNG is used to generate a salt value. This salt is then used when hashing IP addresses and client principals so that they're anonymized. I did not consider this to require a secure RNG, but that was more a hunch then actual reasoning.
@r-birkner perhaps you would have some thoughts on this too?

Reply:
@rikonor, @r-birkner, the `rng_seed` that you are using here is easily predictable, so I'd recommend using an actual (secure) RNG such as `OsRng` (or alternatively at least `ThreadRng` via `rand::thread_rng()` if you have performance concerns). Or do you see a good reason not to do that?

Reply:
Hi @fspreiss, I don't see a good reason not to use a secure rng. I switched to using `OsRng` as you suggested - does it look OK to you now?

Reply:
Yes, looks OK, thanks!